x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/NamedDateTimeProcessorTests.java

```diff
@@ -90,4 +93,22 @@ public void testValidMonthNamesWithNonUTCTimeZone() {
         assertEquals("July", proc.process(new DateTime(10902, 8, 1, 2, 59, DateTimeZone.UTC)));
         assertEquals("August", proc.process(new DateTime(10902, 8, 1, 3, 00, DateTimeZone.UTC)));
     }
+    
+    /*
+     * This method checks the existence of a jvm parameter that should exist in ES jvm.options for Java 9+. If the parameter is
+     * missing, the tests will be skipped. Not doing this, the tests will fail because the day and month names will be in the narrow
+     * format (Mon, Tue, Jan, Feb etc) instead of full format (Monday, Tuesday, January, February etc).
+     * 
+     * Related infra issue: https://github.com/elastic/elasticsearch/issues/33796
+     */
+    private void assumeJava9PlusAndCompatLocaleProviderSetting() {
+        // at least Java 9
+        if (JavaVersion.current().compareTo(JavaVersion.parse("9")) < 0) {
+            return;
+        }
+        String beforeJava9CompatibleLocale = System.getProperty("java.locale.providers");
+        // and COMPAT setting needs to be first on the list
+        boolean isBeforeJava9Compatible = beforeJava9CompatibleLocale != null && beforeJava9CompatibleLocale.split(",")[0].equals("COMPAT");
```

Comment:
What if it's not the first in the list? This piece can be more reliable with something like:
`Arrays.toList(Strings.tokenizeToStringArray(beforeJava9...)).contains("COMPAT")` - works with whitespace, nulls, out-of-order arguments.

Reply:
My understanding is that the list is a preference list with the first provider being the first one checked and used. But I see your point about a more reliable check.

Reply:
Even If it's the case that needs to be first, I agree it would be nice to use the tokenize to trim whitespaces.